xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -771,13 +771,21 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
   // debug
   CLog::Log(LOGDEBUG, "Activating window ID: %i", iWindowID);
 
+  // make sure we check mediasources too
+  if (params.size() > 0)
+    g_passwordManager.strMediasourcePath = params[0];
+
   if (!g_passwordManager.CheckMenuLock(iWindowID))
   {
-    CLog::Log(LOGERROR, "MasterCode is Wrong: Window with id %d will not be loaded! Enter a correct MasterCode!", iWindowID);
+    CLog::Log(LOGERROR,
+              "MasterCode or Mediasource-code is wrong: Window with id %d will not be loaded! "
+              "Enter a correct code!",
+              iWindowID);
     if (GetActiveWindow() == WINDOW_INVALID && iWindowID != WINDOW_HOME)
       ActivateWindow(WINDOW_HOME);
     return;
   }
+  g_passwordManager.strMediasourcePath = "";
```

Comment:
There is a potential return 2 lines above. Unfortunately probably have to duplicate up there as well on that exit path.

There is probably a "cleaner" way to handle it somewhere, but I don't want to keep throwing stuff at you. In general, I think consensus is profiles are in a horrible state and need a large rework which is out of scope of this, so from my viewpoint a good enough approach may be right. And you've solved an issue you have.

Reply:
i would say i‘ll wait for further remarks till tommorrow and then fix the strMediasource-reset and what may come until then. That okay?

Reply:
I'm with @fuzzard, we'll guide you through making a legit PR, and merge when you're happy with the testing you do.